### PR TITLE
boost 1.80.0

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Building `boost-build` fails with
```
    Errno::ENOENT: No such file or directory - boost-build.jam
```
which causes failures in dependents. This line needs adjusting:

https://github.com/Homebrew/homebrew-core/blob/fd1276032151b391c3066c442ba6f9153ae25e74/Formula/boost-build.rb#L29

**Review by carlocab — DISMISSED** [review]
ARM CI looks promising.

**Comment by SMillerDev** [comment]
Don't you jynx it 😅

**Comment by carlocab** [comment]
I didn't jinx `icu4c`!

**Comment by cho-m** [comment]
macOS looks good but we need another run for self-hosted Linux. I expect some extra failures on Linux based on previous PRs.

**Comment by carlocab** [comment]
Formulae modified here are updated pretty quickly. I suggest fixing any remaining Linux failures separately unless they have very many dependents or very many installs (on Linux).

**Comment by cho-m** [comment]
My worry is total number of breakages since last PR #101577 ended up with 70 revision bumps and I think most were added due to Linux. I see that 18 revision bumps went up to `wownero` and then the subsequent commits started back alphabetically at `apngasm`, so there may be up to 52 extra revision bumps needed.

**Comment by carlocab** [comment]
We should restrict any additional revision bumps to those that have lots of installs or have lots of dependents (though the latter will typically imply the former).

Getting the version update shipped to users and done before another PR causes merge conflicts is a higher priority otherwise.

**Comment by carlocab** [comment]
Rebased to pull in Linux CI changes.

**Comment by carlocab** [comment]
I think the `vtk` issue is from trying to do `find_library` on `libdl`.

See https://github.com/conda-forge/assimp-feedstock/issues/14.

**Comment by danielnachun** [comment]
I pushed fixes for `c10t` and `redex` and added revision bumps for the other linkage failures.  The only remaining issues are with `vtk` (still working on that), `pcl` (should be fixed by https://github.com/Homebrew/homebrew-core/pull/110024), and `graph-tool` which needs a revision bump but already has https://github.com/Homebrew/homebrew-core/pull/110016 open which could be merged first or folded into here.

**Comment by cho-m** [comment]
Merged the `pcl` PR, so will need to rebase to pull those changes in before next run.

I think there are more <tt>boost</tt> linkage issues on Linux from last run https://github.com/Homebrew/homebrew-core/runs/8223974076?check_suite_focus=true
<details>
<summary>48 from <tt>brew linkage</tt> failures</summary

- `eigenpy`
- `graph-tool`
- `libtorrent-rasterbar`
- `rdkit`
- `apngasm`
- `avro-cpp`
- `azure-storage-cpp`
- `bastet`
- `collada-dom`
- `cryfs`
- `cryptominisat`
- `dartsim`
- `dspdfviewer`
- `etcd-cpp-apiv3`
- `fastnetmon`
- `fbthrift`
- `field3d`
- `fizz`
- `freeling`
- `fuego`
- `gearman`
- `gnuradio`
- `leela-zero`
- `lgogdownloader`
- `libphonenumber`
- `mapnik`
- `metaproxy`
- `mlpack`
- `mupen64plus`
- `ompl`
- `openvdb`
- `osm2pgrouting`
- `osmium-tool`
- `osrm-backend`
- `pagmo`
- `pcb2gcode`
- `pdns`
- `pktanon`
- `pokerstove`
- `povray`
- `quick-lint-js`
- `scummvm-tools`
- `sfcgal`
- `simgrid`
- `simple-amqp-client`
- `standardese`
- `stp`
- `upscaledb`

</details>

**Comment by iMichka** [comment]
Will need a rebase for `pdnsrec`: #110284

**Comment by carlocab** [comment]
If you need to remove `no long build` conflict to merge another PR, don't add it back. That just blocks other PRs for no good reason. It can be added back once this is rebased.

**Comment by danielnachun** [comment]
Regarding `vtk`, I strongly suspect that similar to some other failures we've seen, it's because one of its dependencies need to be rebuilt.  I am trying to track that down now and open a PR to fix it.  The rest of the linkage failures look like they just need regular revision bumps.  I'm hoping once LLVM ships that we can use the self-host runner to do at least one pass with all the revisions we think we need to make.  We can drop any failures without obvious solutions and rerun and should be able to ship at that point, assuming nothing hugely popular on Linux has any failures.

**Comment by danielnachun** [comment]
I figured it out!  `vtk` was getting the `libdl.so` reference from `netcdf`, which was in turn getting it from `hdf5`.  So `hdf5` needs a revision bump and we should then rebase and try to ship https://github.com/Homebrew/homebrew-core/pull/106520 to update `netcdf` so it is also rebuilt.  After rebuilding those 2 formula I got `vtk` to build, and also got `f3d` to build as well.

**Comment by root42** [comment]
boost 1.79.0_2 upgrade fails as well:

```
> cpp -DNDEBUG builtins.cpp class.cpp command.cpp compile.cpp constants.cpp cwd.cpp debug.cpp debugger.cpp execcmd.
cpp execnt.cpp execunix.cpp filesys.cpp filent.cpp fileunix.cpp frames.cpp function.cpp glob.cpp hash.cpp hcache.cp
p hdrmacro.cpp headers.cpp jam_strings.cpp jam.cpp jamgram.cpp lists.cpp make.cpp make1.cpp md5.cpp mem.cpp modules
.cpp native.cpp object.cpp option.cpp output.cpp parse.cpp pathnt.cpp pathsys.cpp pathunix.cpp regexp.cpp rules.cpp
 scan.cpp search.cpp startup.cpp subst.cpp sysinfo.cpp timestamp.cpp variable.cpp w32_getreg.cpp modules/order.cpp 
modules/path.cpp modules/property-set.cpp modules/regex.cpp modules/sequence.cpp modules/set.cpp -o b2             
cpp: fatal error: too many input files                                                                             
compilation terminated.                                                                                            
> cp b2 bjam                                                                                                       
cp: cannot stat 'b2': No such file or directory                                                                    
                                                                                                                   
Failed to build B2 build engine                                                                                    
                                                                                                                   
READ THIS: https://docs.brew.sh/Troubleshooting                                                                    
                                                                                                                   
These open issues may also help:                                                                                   
boost 1.80.0 https://github.com/Homebrew/homebrew-core/pull/109752                                                 
```
This sent me here. Will this be fixed when boost 1.80 is released and/or will there be a fixed 1.79 formula?

**Comment by carlocab** [comment]
@root42 your problem does not look related to this pull request. Please try the [discussions page](https://github.com/Homebrew/discussions/discussions) if you're looking for more help.

**Comment by danielnachun** [comment]
I pushed revision bumps for everything that appears to need it.  I do need to quickly rebottle `netcdf` so that `vtk` will work and then we shouldn't have any blockers for this, we'll just need the self-hosted Linux runner to be free.

**Comment by cho-m** [comment]
I think we will want to merge `llvm` #106925 and then rebase. Some commits will need to be updated/re-added as they will probably be dropped with simple rebase.

**Comment by danielnachun** [comment]
Yeah that sounds like a plan. `netcdf` is now rebottled so `vtk` should work.  We'll have to see what commits end up needing to be changed.

**Comment by cho-m** [comment]
LLVM is now merged. May also want to get weekly `folly`/etc. bump done (#111143) to avoid delaying that and then try to get `boost` PR done this week before next `folly` update.

File: Formula/redex.rb [hunk 1]
```diff
@@ -106,3 +107,28 @@ def install
...
++++ b/libresource/utils/Vector.h
+@@ -246,12 +246,6 @@ Vector<TYPE>& Vector<TYPE>::operator = (const Vector<TYPE>& rhs) {
+     return *this;
+ }
+
+-template<class TYPE> inline
+-const Vector<TYPE>& Vector<TYPE>::operator = (const Vector<TYPE>& rhs) const {
+-    VectorImpl::operator = (static_cast<const VectorImpl&>(rhs));
+-    return *this;
+-}
```
Comment (cho-m) [inline Formula/redex.rb:134]: Patch is failing. May be due to missing lines after modified code.

**Comment by cho-m** [comment]
`uhd` failed on Monterey ARM but passed on Big Sur ARM:
```
==> Testing uhd
  ==> /opt/homebrew/Cellar/uhd/4.3.0.0/bin/uhd_config_info --version
  dyld[96060]: symbol not found in flat namespace (__ZN5boost10filesystem6detail28directory_iterator_constructERNS0_18directory_iteratorERKNS0_4pathEjPNS_6system10error_codeE)
```

**Comment by carlocab** [comment]
`uhd` just needs a revision bump. It may already be at https://github.com/Homebrew/homebrew-core/pull/111800. `redex` patch needs fixing.

**Comment by danielnachun** [comment]
Sounds good to me.  We already have a PR for the revision bumps on Linux.

**Review by carlocab — DISMISSED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**